GETTING_STARTED.md

```diff
@@ -0,0 +1,374 @@
+  </div>
+)
+
+const TodosContainer = inject((store) => {
+  return {
+    todos: store.todos.getAsArray(),
+    onClick: () => {
+      store.todos.add({ id: Math.random(), title: 'Yo! I am a new todo!' })
+    }
+  }
+})(Todos)
+
+const App = ({ onClick }) => (
+  <TodosContainer />
+)
++
++const listeners = [
++  when('@@krf/ADD>TODOS')(() => {
++    console.log('A todo is added!')
++  })
++]
++
++const AppContainer = listen(listeners)(App)
+
+-export default provider(store)(App)
++export default provider(store)(AppContainer)
+```
+
+Open redux-devtools, click on the button, and look the action is logged into redux-devtools :).\
+The action `@@krf>ADD>TODOS` is triggered when a new todo is added.\
+We then **listen** to this specific action, and **react** to it, here we print a message to the console.
+
+Note that you can bind listeners everywhere you want, this is convenient to add listeners to a screen, so your code logic is near the related screen!
+
+## 7. Connect our app to a HTTP API
+```diff
+import React from 'react'
+import { createStore, types, when } from 'k-ramel'
+import { provider, inject, listen } from '@k-ramel/react'
++import createHttpDriver from '@k-ramel/driver-http'
+
+const store = createStore(
+  {
+    todos: types.keyValue(),
+  },
++  { // a.
++    drivers: {
++      http: createHttpDriver(),
++    },
++  },
+)
+
+store.todos.add({ id: 2, title: 'finish the documentation' })
+store.todos.add({ id: 32, title: 'an other' })
+
+console.log(store.todos.get(2))
+
+const Todos = ({ todos, onClick }) => (
+  <div>
+    <button onClick={onClick}>Add a todo!</button> {/* c. */}
+    <ul>
+      {todos.map(({ id, title }) => (
+        <li key={id}>{title}</li>
+      ))}
+    </ul>
+  </div>
+)
+
+const TodosContainer = inject((store) => {
+  return {
+    todos: store.todos.getAsArray(),
+-    onClick: () => {
+-      store.todos.add({ id: Math.random(), title: 'Yo! I am a new todo!' })
+-    }
++    // b.
++    onClick: () => store.dispatch('@@ui/ADD_TODO>CLICKED')
+  }
+})(Todos)
+
+const App = () => (
+  <TodosContainer />
+)
+
+const listeners = [
+  when('@@krf/ADD>TODOS')(() => {
+    console.log('A todo is added!')
+  }),
++  // c.
++  when('@@ui/ADD_TODO>CLICKED')(async (action, store, drivers) => {
++    const todo = await drivers.http('TODO').post(
++      'https://todo-backend-modern-js.herokuapp.com/todos',
++      {
++        title: 'Yo! I am a new todo!',
++      },
++    )
++
++    store.todos.add(todo)
++  })
+]
+
+const AppContainer = listen(listeners)(App)
+
+export default provider(store)(AppContainer)
+```
+
+First we add a new driver to **k-ramel**, the http driver. (a.) \
+Then we change the `onClick` callback so it dispatchs an empty action whose type is `@@ui/ADD_TODO>CLICKED` (b.), this action is catched into our **listeners**, and we react to it by calling an API (c.).\
+The TODO title is hardcoded, you can see we `await` the API response, and add the todo to the store after the API responds.
```

Comment:
I did this because I think we are more used to `await` a request.
So I try to hook them and refer to what they are used to before going into event driven dev.

Reply:
I understand your idea.
I'm not used to `await` on my work projects :stuck_out_tongue:

Reply:
Because with roadhog-saga you already are in an event driven dev.

But in Cinq, we were using `redux-thunk` and we were doing `await fetch` all the way didn't we?

Reply:
I only use async/await, i guess it depends of people 😄 You have to make a choice